File: Coding_Platforms/Codewars/Python/add_zeros.py
def move_zeros(lst): # Move all zeros to the end

    # create two lists
    # one for non_zeros and another is for zeros
    non_zeros = []
    zeros = []

    for i in lst: # iterate over the list

        if i == 0: # if integer is zero then add in zeros list

            zeros.append(i)
            continue

        else: # if integer is non zero then add in non_zeros list

            non_zeros.append(i)
    
    # return the combination of non zeros and zeros 
    return non_zeros + zeros


print(move_zeros([1, 0, 1, 2, 0, 1, 3]))

File: Coding_Platforms/Codewars/Golang/add_zeros.go
package main

import "fmt"

func main() {

	fmt.Println(MoveZeros([]int{1, 2, 0, 1, 0, 1, 0, 3, 0, 1}))
}

func MoveZeros(arr []int) []int { //Move all zeros to the end

	//create two slice,
	//one is for result silce and another is for store all zeros
	result := []int{}
	zeros := []int{}

	for _, i := range arr { //iterate over the slice

		if i == 0 { //if if interger is zero then add it to zeros slice
			zeros = append(zeros, 0)
			continue
		}
		//append add non-zero integer to the result
		result = append(result, i)
	}

	//finally we have result slice of non-zeros
	//then add zeros slice in the end of result
	result = append(result, zeros...)

	return result
}
